agent/src/beerocks/slave/agent_ucc_listener.cpp

```diff
@@ -186,8 +199,50 @@ bool agent_ucc_listener::handle_start_wps_registration(
         err_string = "invalid WpsConfigMethod, supporting only PBC";
         return false;
     }
-    // TODO trigger WPS PBC on the selected band
-    return true;
+
+    auto freq      = band_to_freq(band->second);
+    auto radio_mac = m_backhaul_manager_ctx.freq_to_radio_mac(freq);
+    if (radio_mac.empty()) {
+        err_string = "Failed to get radio mac for " + band->second;
+        return false;
+    }
+
+    auto it = vaps_map.find(radio_mac);
+    if (it == vaps_map.end()) {
+        err_string = "radio_mac " + radio_mac + " not found";
+        return false;
+    }
+
+    // According to the Multi-AP specification section 5.2, a backhaul STA onboards the agent
+    // using an extension to the WPS PBC onboarding method.
+    // The Agent must advertise WPS on the fronthaul, and during WPS registration
+    // the backhaul credentials are sent in the WPS M8 message.
+    // Therefore, make sure we have fronthaul and backhaul VAPs configured,
+    // then trigger WPS on the first found fAP.
+    sMacAddr fronthaul_bssid = network_utils::ZERO_MAC;
+    sMacAddr backhaul_bssid  = network_utils::ZERO_MAC;
+    // trigger WPS BPC on first fAP bssid
+    for (const auto &vap : it->second.vaps) {
+        if (fronthaul_bssid == network_utils::ZERO_MAC && !vap.backhaul_vap) {
+            fronthaul_bssid = vap.mac;
+        }
+        if (backhaul_bssid == network_utils::ZERO_MAC && vap.backhaul_vap) {
+            backhaul_bssid = vap.mac;
+        }
+    }
+    if (fronthaul_bssid == network_utils::ZERO_MAC) {
+        err_string = "No fAP found for radio_mac " + radio_mac;
+        return false;
+    }
+    if (backhaul_bssid == network_utils::ZERO_MAC) {
+        err_string = "No bAP found for radio_mac " + radio_mac;
+        return false;
+    }
```

Comment:
We shouldn't do this. If PBC is triggered, it is *simultaneously* for fronthaul and backhaul. Therefore, we should trigger it even if no backhaul AP is configured.